wire.md

```diff
@@ -0,0 +1,1008 @@
+   has both a set of *inbound peers* for a request as well as a set of *outbound peers*.
+
+3. A **terminal peer**: a client who received the request from one or
+   more peers and does NOT forward it to any others. A terminal peer has only
+   a set of *inbound peers*.
+
+A peer handling a request who has *outbound peers* (original requester,
+intermediary peer) MUST satisfy any of the following in order to consider a
+request concluded, for each outbound peer:
+
+1. Receives a "no more data" Hash Response (`hash_count = 0`) from the peer.
+2. Sends the peer a Cancel Request, induced by an explicit client action or,
+   for example, a local timeout a client set on the request.
+3. The connection to the peer is lost.
+
+A peer handling a request who has *inbound peers* (intermediary peer, terminal
+peer) MUST satisfy any of the following in order to consider a request
+concluded, for each inbound peer:
+
+1. Sends a "no more data" response back to the peer.
+2. Receives a Cancel Request.
+3. The connection to the peer is lost.
+
+#### 5.2.3 Limits
+Some requests have a `limit` field specifying an upper bound on how many hashes
+a client wishes to receive in response. A peer responding to such a request
+MUST honour that limit by counting how many hashes they send back to the
+requester, including hashes received through other peers that the responding
+client has forwarded that request to.
+
+For example, assume `A` sends a request to `B` with `limit = 50` and `ttl = 1`.
+`B` forwards the request to `C` and `D`. `B` may send back 15 hashes to `A` at
+first, which means there are now a maximum of `50 - 15 = 35` hashes left for `C`
+and `D` combined for `B` to potentially send back.
+
+A requester receiving more than `limit` hashes MAY choose to discard the
+extraneous ones.
+
+#### 5.2.4 Deduplication
+A client who is also an intermediary peer MAY elect to perform deduplication on
+the behalf of a requester, in order to reduce redundant retransmission of post
+or hash data (Post Response and Hash Response, respectively).
+
+For example, consider a client `A` which sends a request to `B` with `ttl = 1`.
+`B` then forwards that request to their peers `C` and `D`. If `B` responds to `A`
+with a set of N hashes or posts `B` could track what they sent. Additionally, in the
+case that `C` or `D`'s responses -- routed through `B` -- contain any
+duplicates that `B` knows were already sent back to `A`, `B` could choose to
+edit these response messages, to omit the duplicates from being needlessly
+retransmitted onwards back to `A`.
+
+Regarding the above section (5.2.3 Limits), hashes that are deduplicated by an
+intermediary peer, and thus not transmitted back to the requester, do not count
+against the `limit`.
+
+An intermediary peer MAY also elect to perform deduplication in the other
+direction on Post Requests, by first checking their local database to see which
+hashes they already have posts for, and editing the original Post Request
```

Comment:
How do they edit the request if it's signed? Or is any peer able to edit any request in transit?

Reply:
- Posts: signed, immutable structures
- Messages (requests & responses): unsigned, transport structures

Sorry, I tried to make it clear in the definitions section that only posts are signed, and the difference between posts and messages. Open to suggestions on making that clearer!